impress-org/givewp
# Feature: update Giving Tuesday banner

Resolves https://app.asana.com/0/1205251546311816/1205651762721240

## Description

This PR includes the Giving Tuesday banner to be displayed for the sale dates 11/20-11/29. During the sale dates the banner should only display on the Give admin screens and all banners remain closed, when any one of the banners on the admin screen is dismissed. 

The banner has different messages and links for users depending on the type of active pricing plan they currently hold. I've included new methods to check the active licenses of the plugins the user has and used that information to determine their pricing tier eligibility. 

Users with Pro plans will not be shown a banner.

## Links:
- Free users: https://go.givewp.com/bf23
- All others (except pro/agency): https://go.givewp.com/bfup23

## Messages:
- Free account users: Upgrade to a Pricing Plan for Recurring Donations, Fee Recovery, and more
- Basic plan users: Upgrade to a Plus Plan to get all must-have add-ons.
- Plus plan users: Upgrade to Pro and get Peer-to-Peer fundraising.

## Affects
All Give admin screens

## Visuals
<img width="1719" alt="Donations list table" src="https://github.com/impress-org/givewp/assets/75056371/2b660f84-68dc-409e-9065-cf6eb44d83c5">
<img width="1719" alt="Donors list table" src="https://github.com/impress-org/givewp/assets/75056371/20a36c50-46df-407e-a98b-058127e280f6">
<img width="1719" alt="Donation Forms list table" src="https://github.com/impress-org/givewp/assets/75056371/ef356037-0230-4390-acdd-5f18f6676cec">
<img width="1719" alt="Settings page" src="https://github.com/impress-org/givewp/assets/75056371/a81a9e1d-b7cb-4b3b-8e3c-029aabde055c">
<img width="1719" alt="Reports page" src="https://github.com/impress-org/givewp/assets/75056371/f8b5f7ad-bf6b-4217-b7af-8c1703668fbd">
<img width="1719" alt="Tools page" src="https://github.com/impress-org/givewp/assets/75056371/e39425d8-03e0-425f-8eb1-671ae677e21a">
<img width="1719" alt="Add form page" src="https://github.com/impress-org/givewp/assets/75056371/8c981db0-cb78-4578-8faf-7b9f9a3becfd">
<img width="1719" alt="Add-ons page" src="https://github.com/impress-org/givewp/assets/75056371/1fa2cc5a-8146-4938-ab2a-1b3bbe5c795a">


## Testing Instructions 

To test :
/src/Promotions/InPluginUpsells/SaleBanners.php 
( line 84 `'startDate' => '2023-11-20 00:00'` can be adjusted to display the banner )


- Verify the banner ONLY displays on our Give admin pages.
- Close one of the banners on any page & verify all the banners remain closed on all pages.
- If you have access to an account with an active license plan - it would be helpful to test that the correct message displays based on the account pricing plan.
- Visit the link to verify it displays the correct page.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205849573153024
